hil-test/tests/uart_tx_rx_async.rs

```diff
@@ -51,4 +64,132 @@ mod tests {
 
         assert_eq!(read, byte);
     }
+
+    #[test]
+    async fn write_async_does_not_return_0(mut ctx: Context) {
+        let bytes = [0; 200];
+
+        for i in 0..50 {
+            let written = ctx.tx.write_async(&bytes).await.unwrap();
+            assert_ne!(written, 0, "Iteration {}", i);
+        }
+    }
+
+    #[test]
+    async fn rx_overflow_is_detected(mut ctx: Context) {
+        let mut to_send: &[u8] = &[0; 250];
+        let mut read = [0u8; 1];
+
+        while !to_send.is_empty() {
+            let written = ctx.tx.write_async(to_send).await.unwrap();
+            to_send = &to_send[written..];
+        }
+        ctx.tx.flush_async().await.unwrap();
+
+        // supposed to panic
+        let res = ctx.rx.read_async(&mut read).await;
+        assert!(matches!(res, Err(RxError::FifoOverflowed)), "{:?}", res);
+    }
+
+    #[test]
+    async fn read_returns_with_partial_data_if_read_times_out(mut ctx: Context) {
+        let mut data = [0u8; 16];
+
+        // We write 4 bytes, but read 16. This should return 4 bytes because the default
+        // RX timeout is not None.
```

Comment:
The default timeout is 10 according to above?

Reply:
it's **not** None.